src/Bicep.Core/TypeSystem/DeclaredTypeManager.cs

```diff
@@ -1408,21 +1408,34 @@ private DeclaredTypeAssignment GetModuleType(ModuleDeclarationSyntax syntax)
 
         private DeclaredTypeAssignment? GetObjectPropertyType(ObjectPropertySyntax syntax)
         {
-            var propertyName = syntax.TryGetKeyText();
+            var propertyName = syntax.TryGetKeyText() ?? (typeManager.GetTypeInfo(syntax.Key) as StringLiteralType)?.RawStringValue;
```

Comment:
Minor: is `RawStringValue` necessary here?

Reply:
`RawStringValue` just hasn't been escaped for inclusion in a Bicep string (i.e., a single quote is still `'` instead of `\'`).